### PR TITLE
Fix binary AUC calculation for boolean labels

### LINKED ISSUE #1029: AUC calculation error on binary classification
ROC AUC calculation fails when labels are boolean. Converting labels to int/float resolves the issue. This behaviour contradicts with scikit-learn and breaks current codebases after patching scikit-learn.

**To Reproduce**
1. Setup the latest `scikit-learn-intelex`.
2. Calculate ROC AUC score calculation on a boolean array.

Alternatively run the following code to reproduce:

```python
from sklearnex import patch_sklearn

patch_sklearn()

import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import roc_auc_score
from sklearn.datasets import make_moons

X, y = make_moons(n_samples=100, random_state=42)
# convert y to binary
y = y.astype(np.bool8)

pred = LogisticRegression(random_state=0).fit(X, y).predict(X)
auc = roc_auc_score(y, pred)
print(f"AUC: {auc:.2f}")
```

**Expected behavior**
Printing AUC score calculation. Scikit-learn handles such boolean inputs under the hood and produces AUC score successfully.

**Output/Screenshots**
```
Intel(R) Extension for Scikit-learn* enabled (https://github.com/intel/scikit-learn-intelex)
Traceback (most recent call last):
  File "/home/ft/Development/intel-test/auc-test.py", line 15, in <module>
    auc = roc_auc_score(y, pred)
  File "/home/ft/miniconda3/envs/sklearnex-dev/lib/python3.9/site-packages/daal4py/sklearn/_device_offload.py", line 83, in wrapper_free
    return wrapper_impl(None, *args, **kwargs)
  File "/home/ft/miniconda3/envs/sklearnex-dev/lib/python3.9/site-packages/daal4py/sklearn/_device_offload.py", line 74, in wrapper_impl
    result = _run_on_device(func, q, obj, *hostargs, **hostkwargs)
  File "/home/ft/miniconda3/envs/sklearnex-dev/lib/python3.9/site-packages/daal4py/sklearn/_device_offload.py", line 65, in _run_on_device
    return dispatch_by_obj(obj, func, *args, **kwargs)
  File "/home/ft/miniconda3/envs/sklearnex-dev/lib/python3.9/site-packages/daal4py/sklearn/_device_offload.py", line 54, in dispatch_by_obj
    return func(*args, **kwargs)
  File "/home/ft/miniconda3/envs/sklearnex-dev/lib/python3.9/site-packages/daal4py/sklearn/metrics/_ranking.py", line 161, in _daal_roc_auc_score
    result = d4p.daal_roc_auc_score(y_true.reshape(-1, 1),
  File "build/daal4py_cy.pyx", line 195, in daal4py._daal4py.daal_roc_auc_score
ValueError: Unsupported NPY type 63 ignored

```

**Environment:**
 - OS: Arch Linux x86_64
 - Compiler: GCC 12.1.1 20220730 
 - Version: 2021.5.0
 - Requirements:

```
daal==2021.6.0
daal4py==2021.6.3
numpy==1.23.1
scikit-learn==1.1.2
scipy==1.9.0
scikit-learn-intelex==2021.6.3

```

### PR DESCRIPTION
**Pull request by furkantektas**
Fixes #1029 

When `y_true` passed as `bool`, it is converted to `int` using `label_binarizer`. But `y_score` was not handled for such case. 

When boolean labels passed, e.g. binary classification problem, AUC calculation fails as  `y_true` is converted to `int` but `y_score` is not. 

This PR converts `y_score` to `int` using `label_binarizer` just like `y_true`.

### CODE REVIEW

**Review by Alexsandruss — APPROVED** [review]

**Comment by Alexsandruss** [comment]
Development branch was changed to `develop`, create PR to this branch, please.

**Comment by furkantektas** [comment]
@Alexsandruss Thanks for the feedback, I have changed the target branch to `develop`.

**Comment by Alexsandruss** [comment]
There are failed sklearn tests from CircleCI:
```
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[roc_auc_score]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[weighted_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[samples_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[micro_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[ovr_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[weighted_ovr_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[ovo_roc_auc]
FAILED miniconda/envs/bld/lib/python3.7/site-packages/sklearn/metrics/tests/test_common.py::test_thresholded_invariance_string_vs_numbers_labels[weighted_ovo_roc_auc]
```

File: daal4py/sklearn/metrics/_ranking.py [hunk 1]
```diff
@@ -158,6 +158,7 @@ def _daal_roc_auc_score(
         if _dal_ready:
             if not np.array_equal(labels, [0, 1]) or labels.dtype == bool:
                 y_true = label_binarize(y_true, classes=labels)[:, 0]
+                y_score = label_binarize(y_score, classes=labels)[:, 0]
```
Comment (Alexsandruss) [inline daal4py/sklearn/metrics/_ranking.py:161]: There is a fix for test fails

```suggestion
                if hasattr(y_score, 'dtype') and y_score.dtype == bool:
                    y_score = label_binarize(y_score, classes=labels)[:, 0]
```

**Comment by furkantektas** [comment]
Thank you @Alexsandruss for the fix.

**Pull request merged**